packages/dev-server/src/esbuild/EsbuildDevServer.ts

```diff
@@ -0,0 +1,220 @@
+import { ExpoConfig } from '@expo/config';
+import { INTERNAL_CALLSITES_REGEX } from '@expo/metro-config';
+import {
+  createDevServerMiddleware,
+  securityHeadersMiddleware,
+} from '@react-native-community/cli-server-api';
+import bodyParser from 'body-parser';
+import type { Server as ConnectServer } from 'connect';
+import * as esbuild from 'esbuild';
+import { ensureDirSync } from 'fs-extra';
+import http from 'http';
+import type { IncomingMessage, ServerResponse } from 'http';
+import { BundleOptions } from 'metro';
+import type Metro from 'metro';
+import path from 'path';
+import resolveFrom from 'resolve-from';
+import { parse as parseUrl } from 'url';
+
+import { BundleOutput, MessageSocket, MetroDevServerOptions } from '../MetroDevServer';
+import { StackFrame } from '../middleware/Symbolicator';
+import clientLogsMiddleware from '../middleware/clientLogsMiddleware';
+import createJsInspectorMiddleware from '../middleware/createJsInspectorMiddleware';
+import { createSymbolicateMiddleware } from '../middleware/createSymbolicateMiddleware';
+import { remoteDevtoolsCorsMiddleware } from '../middleware/remoteDevtoolsCorsMiddleware';
+import { remoteDevtoolsSecurityHeadersMiddleware } from '../middleware/remoteDevtoolsSecurityHeadersMiddleware';
+import { replaceMiddlewareWith } from '../middleware/replaceMiddlewareWith';
+import getBuildOptions from './EsbuildConfig';
+
+const nativeMiddleware = (port: number, _platform: string) => (
```

Comment:
Clean up, focus on JS, maps, assets, and move into a generic function. In theory, this would need to be so good that we could use it across metro and esbuild.